stonesoup/resampler/tests/test_particle.py

```diff
@@ -45,6 +45,17 @@ def test_systematic_even():
                for i, new_particle in enumerate(new_particles))
 
 
+def test_systematic_downsample():
+    particles = [Particle(np.array([[i]]), weight=1/40) for i in range(40)]
+
+    resampler = SystematicResampler()
+
+    new_particles = resampler.resample(particles, 20)
+
+    # Check that the resampler downsamples the particles correctly
+    assert len(new_particles) == 20
+
+
```

Comment:
Would be good to add a test for upsampling as well.

Could even parameterise this test for both up and downsampling

Reply:
The latest commit includes an upsampling test